conan/tools/qbs/qbs.py

```diff
@@ -57,11 +57,23 @@ def add_configuration(self, name, values):
         """
         self._configuration[name] = values
 
+    def _qbs_settings_paths(self):
+        generators_folder = self._conanfile.generators_folder
+        qbs_settings_path = os.path.join(generators_folder, 'qbs_settings.txt')
+        if not os.path.exists(qbs_settings_path):
+            return None, None
+        return os.path.join(generators_folder, 'conan-qbs-settings-dir'), qbs_settings_path
```

Comment:
This settings-dir is something new, I don't fully understand the connection to the QbsProfile.

Reply:
Qbs settings is a collection profiles (as well which profile is default). Profile is similar to Conan profile, just a bunch of toolchain properties.

`$ qbs config --list
defaultProfile: "gcc"
profiles.gcc.cpp.compilerWrapper: "/usr/bin/ccache"
profiles.gcc.cpp.toolchainInstallPath: "/usr/bin"
profiles.gcc.qbs.toolchain: "gcc"
`
Settings are stored in an ini file on win/linux or in a plist file on Mac (in a dir controlled by the --settings-dir flag). Because of that plist, we can't write them directly from Conan (would require a new dependency) so I write settings in an intermediate text file in this dir that is later imported using `qbs config --import` and converted to an inner (ini/plist) format. I suppose I can change the inner format to JSON or use ini on all platforms in Qbs, but that won't work with old Qbs versions anyway; this hacky solution does.